### ISSUE TITLE
Create reuseable text 

### ISSUE THREAD

**Issue by tomhung** [hedgedoc/hedgedoc#1635]
**Is your feature request related to a problem? Please describe.**
I would like to create documents that reuse text in many places throughout. Adding it into a YAML top section would allow the reuse strings / markdown section.

**Describe the solution you'd like**
In the top section allows syntax like the following:

```
---
reuse:
  test_str: This is a sentence that will get reused. 
  test_img: ![](https://notes.example.com/uploads/upload_35f3deafcca6012efb4effb94e3c4280.png =600x)
  test_md: > this is a blockquote [color=blue]
---
```

in the markdown body it can be referenced like the following

```
# Test Document

This is the first sentence. [reuse:test_str] This is the third sentence.

I can reuse the sentence again. Like here [reuse:test_str]

## Images

[reuse:test_img]

## markdown

[reuse:test_md]
```

**Describe alternatives you've considered**
I have looked in the V2 demo for this feature. I have evaluated many markdown editors and Hedgedoc provides me with the features I most need.

**Additional context**
I commonly create templates for notes. It would speed up completion of my notes if I generally could just fill in the top YAML. I also print-to-PDF and send the notes to clients.

This would be similar to [reference] and [link text][reference] but allow for more general markdown reuse

**Comment by SISheogorath** [comment]
I don't think that HedgeDoc should become a templating engine. That's what eh… templating engines are for.

And it'll basically just snowball potential exploits. [Look at XML templates and how that worked out…](https://en.wikipedia.org/w/index.php?title=Billion_laughs_attack&oldid=1024730340)

Finally, your example with the blockquote might end up very unintentional as the `>` sign is a [reserved character in YAML for line-folding](https://yaml.org/spec/1.2/spec.html#id2779048)

All in all, I don't see how this will really benefit HedgeDoc as a markdown **editor**, while I see a lot of risks associated with it.
